src/java.base/share/classes/java/util/UUID.java

```diff
@@ -99,24 +104,190 @@ public final class UUID implements java.io.Serializable, Comparable<UUID> {
     /*
      * The random number generator used by this class to create random
      * based UUIDs. In a holder class to defer initialization until needed.
+     * <p>
+     * The implementation buffers the reads to avoid bashing SecureRandom with
+     * small requests. It also maintains a SecureRandom per buffer to alleviate
+     * scalability bottlenecks when reading from a single (synchronized) SecureRandom.
      */
-    private static class Holder {
-        static final SecureRandom numberGenerator = new SecureRandom();
+    private static final class RandomUUID {
+        static final String PROP_NAME_PRNG_NAME = "java.util.UUID.prngName";
+        static final String PROP_NAME_BUF_COUNT = "java.util.UUID.buffersCount";
+
+        static final String PRNG_NAME;
+        static final int BUFS_COUNT;
+        static final Buffer[] BUFS;
+
+        private static final int roundPowerOfTwo(int x) {
+            int n = -1 >>> Integer.numberOfLeadingZeros(x - 1);
+            return (n < 0) ? 1 : (n + 1);
+        }
+
+        static {
+            try {
+                PRNG_NAME = System.getProperty(PROP_NAME_PRNG_NAME, null);
```

Comment:
You'll have to use GetPropertyAction.privilegedGetProperty as this will otherwise fail if someone runs with a SM.

Reply:
Good spot. Should be fixed now.